core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/function/stringfunction/SubstringExpression.kt

```diff
@@ -19,13 +19,13 @@ class SubstringExpression(
                 symbol = "SUBSTR",
                 inStrDopeQuery,
                 startPos.toDopeType().toDopeQuery(manager),
-                length.toDopeType().toDopeQuery(manager),
+                length?.toDopeType()?.toDopeQuery(manager),
             ),
             parameters = inStrDopeQuery.parameters,
         )
     }
 }
 
-fun substr(inStr: TypeExpression<StringType>, startPos: Int, length: Int) = SubstringExpression(inStr, startPos, length)
+fun substr(inStr: TypeExpression<StringType>, startPos: Int, length: Int? = null) = SubstringExpression(inStr, startPos, length)
```

Comment:
can you add a test, which covers this new functionality?

Reply:
done